File: crashpecker/src/main/java/me/peace/provider/WoodPeckerProvider.java
package me.peace.provider;

import android.content.ContentProvider;
import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import me.peace.communication.ProcessConstant;
import me.peace.communication.ProcessCoordinator;

public class WoodPeckerProvider extends ContentProvider {

    private ProcessCoordinator coordinator;

    @Override
    public boolean onCreate() {
        coordinator = new ProcessCoordinator();
        return true;
    }

    
    @Nullable
    @Override
    public Cursor query(@NonNull Uri uri,  @Nullable String[] projection,  @Nullable String selection,  @Nullable String[] selectionArgs,  @Nullable String sortOrder) {
        return null;
    }

    
    @Nullable
    @Override
    public String getType(@NonNull Uri uri) {
        return null;
    }

    
    @Nullable
    @Override
    public Uri insert(@NonNull Uri uri,  @Nullable ContentValues values) {
        return null;
    }

    @Override
    public int delete(@NonNull Uri uri,  @Nullable String selection,  @Nullable String[] selectionArgs) {
        return 0;
    }

    @Override
    public int update(@NonNull Uri uri,  @Nullable ContentValues values,  @Nullable String selection,  @Nullable String[] selectionArgs) {
        return 0;
    }

    
    @Nullable
    @Override
    public Bundle call( @NonNull String method,  @Nullable String arg,  @Nullable Bundle extras) {
        if (ProcessConstant.METHOD_SAVE_SP.equals(method)) {
            Context context = getContext();
            coordinator.saveSharedPreferences(context,arg,extras);
        } else if (ProcessConstant.METHOD_LOAD_SP.equals(method)) {
            Context context = getContext();
            return coordinator.loadSharedPreferences(context,arg,extras);
        }
        return null;
    }
}

File: crashpecker/src/main/java/me/peace/provider/WoodPeckerProvider.kt
package me.peace.provider


import android.content.ContentProvider
import android.content.ContentValues
import android.database.Cursor
import android.net.Uri
import android.os.Bundle
import me.peace.communication.ProcessConstant
import me.peace.communication.ProcessCoordinator

class WoodPeckerProvider : ContentProvider() {
    private var coordinator: ProcessCoordinator? = null
    override fun onCreate(): Boolean {
        coordinator = ProcessCoordinator()
        return true
    }

    override fun query(
        uri: Uri,
        projection: Array<String>?,
        selection: String?,
        selectionArgs: Array<String>?,
        sortOrder: String?
    ): Cursor? {
        return null
    }

    override fun getType(uri: Uri): String? {
        return null
    }

    override fun insert(uri: Uri, values: ContentValues?): Uri? {
        return null
    }

    override fun delete(uri: Uri, selection: String?, selectionArgs: Array<String>?): Int {
        return 0
    }

    override fun update(
        uri: Uri,
        values: ContentValues?,
        selection: String?,
        selectionArgs: Array<String>?
    ): Int {
        return 0
    }

    override fun call(method: String, arg: String?, extras: Bundle?): Bundle? {
        if (ProcessConstant.METHOD_SAVE_SP == method) {
            val context = context
            coordinator!!.saveSharedPreferences(context, arg, extras)
        } else if (ProcessConstant.METHOD_LOAD_SP == method) {
            val context = context
            return coordinator!!.loadSharedPreferences(context, arg, extras)
        }
        return null
    }
}